crane/template.go

```diff
@@ -0,0 +1,131 @@
+package crane
+
+import (
+	"fmt"
+	"github.com/michaelsauter/crane/print"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+type TemplateInfo struct {
+	TargetedContainers []ContainerInfo
+	DependencyMap      DependencyGraph
+	Groups             map[string][]string
+}
+type TemplateInfoForContainer struct {
+	CurrentContainer   ContainerInfo
+	TargetedContainers []ContainerInfo
+	DependencyMap      DependencyGraph
+	Groups             map[string][]string
+}
+
+// Sub-interface of Container
+type ContainerInfo interface {
+	Name() string
+	Dockerfile() string
+	Image() string
+	Id() string
+	Dependencies() *Dependencies
```

Comment:
I know this makes life easier when generating a per-container systemd/upstart file, but I feel like this shouldn't be exposed as-is here. I see `ContainerInfo` as something providing either raw configuration, or state about that container. `Dependencies` is an opinionated transformation of  links/volumesFrom/net, and is already available via a lookup in the `DependencyMap`. If we need access to raw dependencies, maybe we could expose `RunParams()` ? I know it seems clunky but what if we need access to the exposed ports from the template?